src/assets/src/services/api.ts

```diff
@@ -169,13 +180,19 @@ export const setStatus = async (queue_id: number, open: boolean) => {
     return await resp.json();
 }
 
-export const getMyUser = async (user_id: number) => {
-    const resp = await fetch(`/api/users/${user_id}/`, { method: "GET" });
+const getUser = async (identifier: number | string) => {
+    const resp = await fetch(`/api/users/${identifier}/`, { method: "GET" });
     await handleErrors(resp);
-    return await resp.json() as MyUser;
+    return await resp.json() as User | MyUser;
 }
 
-export const updateMyUser = async (user_id: number, phone_number: string) => {
+export const getUserById = async (user_id: number) =>
+    getUser(user_id);
+
+export const getUserByUniqname = async (uniqname: string) =>
+    getUser(uniqname);
```

Comment:
Nit-pick: are these necessary?

Reply:
I compressed these into `getUser` by renaming its first param to `id_or_username`. I habitually avoid `this_or_that` param names as they can scale poorly with number of options or number of parameters, but in this case I can't imagine more than these two.